service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/storageFolder/ControlledAwsStorageFolderResource.java

```diff
@@ -0,0 +1,213 @@
+  }
+
+  public String getBucketName() {
+    return bucketName;
+  }
+
+  public String getPrefix() {
+    return prefix;
+  }
+
+  // -- getters not included in serialization --
+
+  @Override
+  @JsonIgnore
+  public WsmResourceType getResourceType() {
+    return WsmResourceType.CONTROLLED_AWS_STORAGE_FOLDER;
+  }
+
+  @Override
+  @JsonIgnore
+  public WsmResourceFamily getResourceFamily() {
+    return WsmResourceFamily.AWS_STORAGE_FOLDER;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  @JsonIgnore
+  public Optional<UniquenessCheckAttributes> getUniquenessCheckAttributes() {
+    return Optional.of(
+        new UniquenessCheckAttributes()
+            .uniquenessScope(UniquenessScope.GLOBAL)
+            .addParameter("prefix", prefix));
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void addCreateSteps(
+      CreateControlledResourceFlight flight,
+      String petSaEmail,
+      AuthenticatedUserRequest userRequest,
+      FlightBeanBag flightBeanBag) {
+    RetryRule cloudRetry = RetryRules.cloud();
+    flight.addStep(
+        new CreateAwsStorageFolderStep(
+            this, flightBeanBag.getAwsCloudContextService(), userRequest),
+        cloudRetry);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void addDeleteSteps(DeleteControlledResourcesFlight flight, FlightBeanBag flightBeanBag) {
+    flight.addStep(
+        new DeleteAwsStorageFolderStep(this, flightBeanBag.getAwsCloudContextService()),
+        RetryRules.cloud());
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void addUpdateSteps(UpdateResourceFlight flight, FlightBeanBag flightBeanBag) {
+    throw new ApiException("addUpdateSteps NotImplemented");
+  }
+
+  public ApiAwsStorageFolderAttributes toApiAttributes() {
+    return new ApiAwsStorageFolderAttributes()
+        .bucketName(getBucketName())
+        .prefix(getPrefix())
+        .region(getRegion());
+  }
+
+  public ApiAwsStorageFolderResource toApiResource() {
+    return new ApiAwsStorageFolderResource()
+        .metadata(super.toApiMetadata())
+        .attributes(toApiAttributes());
+  }
+
+  @Override
+  public String attributesToJson() {
+    return DbSerDes.toJson(
+        new ControlledAwsStorageFolderAttributes(getBucketName(), getPrefix(), getRegion()));
+  }
+
+  @Override
+  public ApiResourceAttributesUnion toApiAttributesUnion() {
+    ApiResourceAttributesUnion union = new ApiResourceAttributesUnion();
+    union.awsStorageFolder(toApiAttributes());
+    return union;
+  }
+
+  @Override
+  public void validate() {
+    super.validate();
+    if (getResourceType() != WsmResourceType.CONTROLLED_AWS_STORAGE_FOLDER
+        || getResourceFamily() != WsmResourceFamily.AWS_STORAGE_FOLDER
+        || getStewardshipType() != StewardshipType.CONTROLLED) {
+      throw new InconsistentFieldsException("Expected CONTROLLED_AWS_STORAGE_FOLDER");
+    }
+    if ((getBucketName() == null) || (getPrefix() == null) || (getRegion() == null)) {
```

Comment:
move the folder name validation from controller to here

Reply:
also nit: rm the unnecessary ()

Reply:
Retaining () to aid readability